drivers/i2c/Kconfig

```diff
@@ -218,7 +220,7 @@ config I2C_1
 config I2C_1_NAME
 	string "Port 1 device name"
 	default "I2C_1"
-	depends on I2C_1
+	depends on I2C_1 && !HAS_DTS_I2C
 
 config I2C_1_DEFAULT_CFG
```

Comment:
I left all I2C_n_DEFAULT_CFG in Kconfig.defconfig files for now. As we plan not to handle the default configuration in the DTS file, I believe we should address this issue in another PR.